### PR TITLE
 programs/kde-pim: init 

### LINKED ISSUE #299152: Add option "programs.kde-pim.enable"
## Problem

There is currently no documentation, on how to combine the different PIM libraries, and what is needed to get a full Kontact experience. 

I installed all the seemingly relevant libraries, and it is still resulting in an error. 

I question the sense of shipping Kontact without Akonadi.

## Proposal

Document on how to combine the different PIM libraries, and clearly stating, which one is needed in which case, and what combinations lead to what outcome. 

## Checklist

- [x] checked [latest Nixpkgs manual] \([source][nixpkgs-source]) and [latest NixOS manual] \([source][nixos-source])
- [x] checked [open documentation issues] for possible duplicates
- [x] checked [open documentation pull requests] for possible solutions

[latest Nixpkgs manual]: https://nixos.org/manual/nixpkgs/unstable/
[latest NixOS manual]: https://nixos.org/manual/nixos/unstable/
[nixpkgs-source]: https://github.com/NixOS/nixpkgs/tree/master/doc
[nixos-source]: https://github.com/NixOS/nixpkgs/tree/master/nixos/doc/manual
[open documentation issues]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+label%3A%229.needs%3A+documentation%22
[open documentation pull requests]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+is%3Apr+label%3A%228.has%3A+documentation%22%2C%226.topic%3A+documentation%22

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by SuperSandro2000**
## Description of changes

Closes #299152

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by ShalokShalom** [comment]
In order to clarify the open question about how to install Kontact, Merkuro and standalone KMail etc: 

Currently, it is planned to have them listed normally, and not activated by the service proposed here?

I think that would be a deviation from how things work in NixOS today.
What are the reasons for this?

I would rather choose between them as options of the service, as used to in other cases.

File: nixos/modules/module-list.nix [hunk 1]
```diff
@@ -206,6 +206,7 @@
   ./programs/java.nix
   ./programs/joycond-cemuhook.nix
   ./programs/k3b.nix
+  ./programs/kde-pin.nix
```
[the remark this answers is not in the document]
Comment (ShalokShalom) [reply nixos/modules/module-list.nix:209]: And the module itself was also named so: 
https://github.com/NixOS/nixpkgs/pull/299599/files#diff-7090d10b3fcebb7a872a0eba437055984b52678ba6af48a164dc1ec83f2f1ac4

**Comment by ShalokShalom** [comment]
@SuperSandro2000 Does `enabling programs.kde-pim by default` mean Kontact will be installed on any Plasma 6 session, unless explicitly excluded?

**Comment by SuperSandro2000** [comment]
It won't. I've added explicit options for kmail, kontact and mercuro in my latest push. I am not sure if that is the final design though.

**Review by Shawn8901 — APPROVED** [review]
running that on my system since some days with `kde-pim.enable` and `kde-pim.kmail`, seems to work fine, so lgtm

**Comment by Redhawk18** [comment]
Is there anything I can do to help, after searching the same issues I've found this pr that seems to be stuck.

**Comment by Shawn8901** [comment]
if you are ok with the diff, approving the PR could be a good start :slightly_smiling_face:

**Comment by Redhawk18** [comment]
lgtm

**Comment by Shawn8901** [comment]
fyi: there is an approval function in github that i meant.

**Comment by Redhawk18** [comment]
> fyi: there is an approval function in github that i meant.

I dont have any kinds of perms to do that

**Comment by Gerg-L** [comment]
> I dont have any kinds of perms to do that

Don't need any

**Review by Gerg-L — APPROVED** [review]
LGTM

**Review by Redhawk18 — APPROVED** [review]

File: nixos/modules/programs/kde-pim.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    merkuro = lib.mkEnableOption "Merkuro";
+  };
+
+  config = lib.mkIf cfg.enable {
+    environment.systemPackages = with pkgs.kdePackages; [
+      # core packages
+      akonadi
+      kdepim-runtime
+    ] ++ lib.optionals cfg.kmail [
+      akonadiconsole
```
Comment (bkchr) [inline nixos/modules/programs/kde-pim.nix:20]:
```suggestion
      akonadiconsole
      akonadi-search
```

The search is required for indexing the mails.

**Comment by K900** [comment]
Backporting because people keep hitting this.

**Pull request merged**